src/content.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for checking conflicts using 3-way merge and create content based on a 3 way merge."""
+
+import difflib
+import tempfile
+from pathlib import Path
+
+from git.exc import GitCommandError
+from git.repo import Repo
+
+from .exceptions import ContentError
+
+
+def conflicts(base: str, theirs: str, ours: str) -> str | None:
+    """Check for merge conflicts based on the git merge algorithm.
+
+    Args:
+        base: The starting point for both changes.
+        theirs: The other change.
+        ours: The local change.
+
+    Returns:
+        The description of the merge conflicts or None if there are no conflicts.
+    """
+    try:
+        merge(base=base, theirs=theirs, ours=ours)
+    except ContentError as exc:
+        return str(exc)
+    return None
+
+
+def merge(base: str, theirs: str, ours: str) -> str:
```

Comment:
The reason is that the classes, like `ContentChange`, are more intended to be used to pass data around and define interfaces instead of carrying business logic with them. This repo has generally been functional programming oriented rather than OOP, although in a few cases where it really makes sense OOP is used, typically in cases where a property would just be a transformation of the data which can be a nice thing to add to the data

Reply:
Functional and OOP are not really antithetic, in the sense that they can co-exists (e.g. scala, switf, rust). I personally believe it is fine to have domain objects/data classes (that contain data) with helpers methods attached to it. But I understand this is a matter of taste, and although I felt worth suggesting, it is really up to you decide whether you like this, without your feeling that it is ripping apart the style and the structure of the code